### ISSUE TITLE
CIL.Expressions: debug mode problems

### ISSUE THREAD

**Issue by exyi** [naasking/CIL#2]
When I compile my project in Debug, the compiler emits some redundant IL like `... goto NextLine; NextLine: ...` which breaks the decompiler.

**Comment by naasking** [comment]
Yes, the control-flow analysis is incomplete at the moment, in that it doesn't properly merge branches in all cases. I'll try to get to it this weekend.

**Comment by exyi** [comment]
I understand where is the problem, I was also writing a "decompiler" :D Just putting it here to make sure that it's a known problem.
